app/courses/[courseId]/scoreboard/page.tsx

```diff
@@ -0,0 +1,160 @@
+"use client";
+import * as React from "react";
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TablePagination from "@mui/material/TablePagination";
+import TableRow from "@mui/material/TableRow";
+import { useParams, useRouter, useSearchParams } from "next/navigation";
+import { graphql, useLazyLoadQuery } from "react-relay";
+import { pageScoreboardQuery } from "@/__generated__/pageScoreboardQuery.graphql";
+import { Button } from "@mui/material";
+import Link from "next/link";
+
+interface Column {
+  id: "place" | "name" | "power";
+  label: string;
+  minWidth?: number;
+  align?: "right" | "center";
+}
+
+const columns: readonly Column[] = [
+  { id: "place", label: "Place", minWidth: 100 },
+  { id: "name", label: "Name", minWidth: 200, align: "center" },
+  {
+    id: "power",
+    label: "Power",
+    minWidth: 100,
+    align: "right",
+  },
+];
+
+interface Data {
+  place: string;
+  name: string;
+  power: number;
+}
+
+function createData(place: string, name: string, power: number): Data {
+  return { place, name, power };
+}
+
+function capitalizeFirstLetter(str: string): string {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+export default function StickyHeadTable() {
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(10);
+
+  const params = useParams();
+  const courseId = params.courseId;
+
+  const { scoreboard } = useLazyLoadQuery<pageScoreboardQuery>(
+    graphql`
+      query pageScoreboardQuery($courseId: UUID!) {
+        scoreboard(courseId: $courseId) {
+          user {
+            userName
+          }
+          powerScore
+        }
+      }
+    `,
+    { courseId: courseId }
+  );
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRowsPerPage(+event.target.value);
+    setPage(0);
+  };
+
+  const rows: any[] = [];
+
+  scoreboard.forEach((element, index: number) => {
+    index++;
+    rows.push(
+      createData(
+        `${index}`,
+        capitalizeFirstLetter(element.user.userName),
+        element.powerScore
+      )
+    );
+  });
```

Comment:
das würde ich eher so lösen:

```suggestion
  const rows: Data[] = scoreboard.map((element, index) => createData(
        `${index}`,
        capitalizeFirstLetter(element.user.userName),
        element.powerScore
      ));
```